c++/src/OrcHdfsFile.cc

```diff
@@ -66,22 +64,22 @@ namespace orc {
         options = config->GetOptions();
       }
       hdfs::IoService * io_service = hdfs::IoService::New();
-      //Wrapping fs into a shared pointer to guarantee deletion
-      std::shared_ptr<hdfs::FileSystem> fs(hdfs::FileSystem::New(io_service, "", options));
-      if (!fs) {
+      //Wrapping file_system into a unique pointer to guarantee deletion
+      file_system = std::unique_ptr<hdfs::FileSystem>(hdfs::FileSystem::New(io_service, "", options));
+      if (!file_system) {
```

Comment:
Unfortunately unique_ptr can be redefined as auto_ptr if platform doesn't support unique_ptr (see orc_config.hh). We talked about removing this redefine, but it hasn't been done yet. So I'd suggest we stick to file_system.get() != nullptr for now.